Land stubs for the Screen Orientation API

cdumez opened the pull request:
#### 4747b6ca4137255b0bb05230d53bf701159e4511
<pre>
Land stubs for the Screen Orientation API
<a href="https://bugs.webkit.org/show_bug.cgi?id=245804">https://bugs.webkit.org/show_bug.cgi?id=245804</a>

Reviewed by Brent Fulgham.

Land stubs for the Screen Orientation API:
- <a href="https://w3c.github.io/screen-orientation">https://w3c.github.io/screen-orientation</a>

* Source/WTF/Scripts/Preferences/WebPreferencesExperimental.yaml:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/dom/EventTargetFactory.in:
* Source/WebCore/page/Screen.cpp:
(WebCore::Screen::orientation):
* Source/WebCore/page/Screen.h:
* Source/WebCore/page/Screen.idl:
* Source/WebCore/page/ScreenOrientation.cpp: Added.
(WebCore::ScreenOrientation::create):
(WebCore::ScreenOrientation::ScreenOrientation):
(WebCore::ScreenOrientation::document const):
(WebCore::ScreenOrientation::lock):
(WebCore::ScreenOrientation::unlock):
(WebCore::ScreenOrientation::type const):
(WebCore::ScreenOrientation::angle const):
(WebCore::ScreenOrientation::activeDOMObjectName const):
(WebCore::ScreenOrientation::virtualHasPendingActivity const):
(WebCore::ScreenOrientation::eventListenersDidChange):
* Source/WebCore/page/ScreenOrientation.h: Added.
* Source/WebCore/page/ScreenOrientation.idl: Added.
* Source/WebCore/page/ScreenOrientationLockType.h: Added.
* Source/WebCore/page/ScreenOrientationType.h: Added.
* Source/WebCore/page/csp/CSPViolationReportBody.cpp:

Canonical link: <a href="https://commits.webkit.org/255032@main">https://commits.webkit.org/255032@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3507140d8851b63bd4968e86609fd3039a7af2c9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/90982 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/35553 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/21519 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/100328 "Built successfully") | [❌ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/158833 "Found 1 new test failure: fast/dom/Window/window-appendages-cleared.html (failure)") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/34054 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/29071 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/83328 "Built successfully") | [💥 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/97120 "An unexpected error occured. Recent messages:Failed to print configuration") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/96638 "Passed tests") | [⏳ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/iOS-16-Simulator-WK2-Tests-EWS "Waiting to run tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/77769 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/26945 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/81887 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/81613 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/69980 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/35134 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/15633 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/32934 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/16620 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3502 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/36712 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/39581 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/38639 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/35704 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

marcoscaceres commented on Source/WebCore/page/csp/CSPViolationReportBody.cpp:
```diff
@@ -30,6 +30,7 @@
 #include "FormData.h"
 #include "SecurityPolicyViolationEvent.h"
 #include <wtf/IsoMallocInlines.h>
+#include <wtf/JSONValues.h>
```
maybe stray change?

cdumez replied:
Intentional change to make sure the patch builds. Because of the way WebKit is built, whenever you add new files to the project, you end up finding missing includes and causing build failures. This is a bit annoying but supposedly it makes the build faster.

brentfulgham reviewed (approved):
LGTM. r=me

cdumez committed:
Land stubs for the Screen Orientation API
https://bugs.webkit.org/show_bug.cgi?id=245804

Reviewed by Brent Fulgham.

Land stubs for the Screen Orientation API:
- https://w3c.github.io/screen-orientation

* Source/WTF/Scripts/Preferences/WebPreferencesExperimental.yaml:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources-output.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/WebCoreBuiltinNames.h:
* Source/WebCore/dom/EventTargetFactory.in:
* Source/WebCore/page/Screen.cpp:
(WebCore::Screen::orientation):
* Source/WebCore/page/Screen.h:
* Source/WebCore/page/Screen.idl:
* Source/WebCore/page/ScreenOrientation.cpp: Added.
(WebCore::ScreenOrientation::create):
(WebCore::ScreenOrientation::ScreenOrientation):
(WebCore::ScreenOrientation::document const):
(WebCore::ScreenOrientation::lock):
(WebCore::ScreenOrientation::unlock):
(WebCore::ScreenOrientation::type const):
(WebCore::ScreenOrientation::angle const):
(WebCore::ScreenOrientation::activeDOMObjectName const):
(WebCore::ScreenOrientation::virtualHasPendingActivity const):
(WebCore::ScreenOrientation::eventListenersDidChange):
* Source/WebCore/page/ScreenOrientation.h: Added.
* Source/WebCore/page/ScreenOrientation.idl: Added.
* Source/WebCore/page/ScreenOrientationLockType.h: Added.
* Source/WebCore/page/ScreenOrientationType.h: Added.
* Source/WebCore/page/csp/CSPViolationReportBody.cpp:

Canonical link: https://commits.webkit.org/255032@main
